proxy/src/main.rs

```diff
@@ -33,13 +36,57 @@ pub struct ProxyConf {
 
     /// control plane address where we would check auth.
     pub cplane_address: SocketAddr,
+
+    pub ssl_config: Option<Arc<rustls::ServerConfig>>,
 }
 
 pub struct ProxyState {
     pub conf: ProxyConf,
     pub waiters: Mutex<HashMap<String, mpsc::Sender<anyhow::Result<DatabaseInfo>>>>,
 }
 
+fn configure_ssl(arg_matches: &ArgMatches) -> anyhow::Result<Option<Arc<rustls::ServerConfig>>> {
```

Comment:
This function is pretty visually messy. A couple suggestions below. Feel free to use them or leave them as you see fit :)